perfkitbenchmarker/scripts/database_scripts/launch_mysql_service.py

```diff
@@ -0,0 +1,261 @@
+    sys.exit(1)
+  run_uri = FLAGS.run_uri
+  if not run_uri:
+    logging.info('No run_uri given. Will run full mysql_service_benchmark '
+                 'test.')
+    run_uri = _provision_prepare_pkb()
+    logging.info('Provision and prepare completed. Run uri assigned: %s',
+                 run_uri)
+  _run(run_uri)
+  if not FLAGS.run_uri:
+    _cleanup_teardown_pkb(run_uri)
+
+
+def _provision_prepare_pkb():
+  """Run provision and prepare stage of PKB benchmark.
+
+  Returns:
+    run_uri: (string)
+  """
+  pkb_cmd = PKB + STAGE_FLAG + PROVISION + ',' + PREPARE
+  if FLAGS.additional_flags:
+    pkb_cmd = _append_additional_flags(pkb_cmd)
+  # PKB run with prepare,provision, wait
+  logging.info('Provision and prepare sysbench with the following command:\n%s',
+               pkb_cmd)
+  [stdout_filename, stderr_filename] = _generate_filenames(PROVISION, None)
+  _execute_pkb_cmd(pkb_cmd, stdout_filename, stderr_filename)
+  return _get_run_uri(stderr_filename)
+
+
+def _run(run_uri):
+  """Run stage of PKB benchmark.
+
+  Args:
+    run_uri: (string).
+  """
+
+  run_iterations = len(FLAGS.thread_count_list)
+  logging.info(
+      'Beginning run phase. Will execute runs with %d different thread counts.',
+      run_iterations)
+  for t in FLAGS.thread_count_list:
+    pkb_cmd = (PKB + STAGE_FLAG + RUN + URI_FLAG + run_uri + THREAD_FLAG +
+               str(t) + RUN_TIME + str(FLAGS.sysbench_run_seconds) + ' &')
+    if FLAGS.additional_flags:
+      pkb_cmd = _append_additional_flags(pkb_cmd)
+    stdout_filename, stderr_filename = _generate_filenames(RUN, t)
+    logging.info('Executing PKB run with thread count: %i', t)
+    _execute_pkb_cmd(pkb_cmd, stdout_filename, stderr_filename)
+#     _wait_for_run(stderr_filename)
+    logging.info('Finished executing PKB with thread count: %i', t)
+    time.sleep(SLEEP_TIME_BETWEEN_RUNS)
+
+
+def _cleanup_teardown_pkb(run_uri):
+  """Run cleanup stage of PKB benchmark.
+
+  Args:
+    run_uri: (string)
+  """
+  logging.info('Run phase complete. Starting cleanup/teardown.')
+  pkb_cmd = (PKB + STAGE_FLAG + CLEANUP + ',' + TEARDOWN + URI_FLAG + run_uri)
+  [stdout_filename, stderr_filename] = _generate_filenames(CLEANUP, None)
+  _execute_pkb_cmd(pkb_cmd, stdout_filename, stderr_filename)
+#   _wait_for_run(stderr_filename)
+  logging.info('Finished executing PKB cleanup and teardown.')
+
+
+def _execute_pkb_cmd(pkb_cmd, stdout_filename, stderr_filename):
+  """Given pkb run command, execute.
+
+  Args:
+    pkb_cmd: (str)
+    stdout_filename: (str) filename string.
+    stderr_filename: (str) filename_str
+  """
+  stdout_file = open(stdout_filename, 'w+')
+  stderr_file = open(stderr_filename, 'w+')
+  pkb_cmd_list = shlex.split(pkb_cmd)
+  logging.info('pkb command list: %s', str(pkb_cmd_list))
+  p = subprocess.Popen(pkb_cmd_list, stdout=stdout_file, stderr=stderr_file)
+  logging.info('Waiting for PKB call to finish.')
+  p.wait()  # add timeout time
```

Comment:
The question is about timeout value for the p.wait() command. Is there a way to specify a timeout  for p.wait()? (Note: the sleep in-between the individual runs is not relevant to this discussion). 

This discussion is more about how we guard against a run that's hung/take too long. Please at least take a look into this, if it's simple to add a timeout, we can do it now in this PR; otherwise, we can add a TODO here and address it later.